src/libostree/ostree-repo-libarchive.c

```diff
@@ -81,13 +82,23 @@ path_relative (const char *src)
   if (src[0] == '.' && src[1] == '\0')
     src += 1;
 
+  /* make sure that the final path is valid (no . or ..) */
+  if (!ot_util_path_split_validate (src, NULL, error))
```

Comment:
A bit expensive to do it this way, we end up mallocing all the path components just to throw them away.  But eh...we can revisit this later.